watek_glowny.c

```diff
@@ -2,47 +2,56 @@
 #include "watek_glowny.h"
 
 int i;
-void mainLoop()
+
+
+void getEntryPermission()
 {
-    srandom(rank);
-    while (state != InFinish) {
-        /* code here */
-        if (state == Mission) {
-            sleep(rand() % 5);
-            changeState(EnteringBar);
-            println("Czekam na bar");
-        } else if (state == EnteringBar) {
-            pthread_mutex_lock(&lamportMut);
-            sem_init(&ackSemaphore, FALSE, 0);
+	sem_init(&ackSemaphore, FALSE, 0);
             packet_t packet;
             packet.ts = ++lamportClock;
             for (i = 0; i < size; i++) {
                 if (i != rank) {
                     sendPacket(&packet, i, REQ);
                 }
                 ackVector[i] = FALSE;
-            }            
+            }
             barEntrancePriority = priorityFunc(packet);
-	        debug("Czekam na %d x ACK moj priorytet %d", barSize - size + 2, barEntrancePriority);
+                debug("Czekam na %d x ACK moj priorytet %d", barSize - size + 2, barEntrancePriority);
             pthread_mutex_unlock(&lamportMut);
-	    for (i = 0; i < size - barSize; i++)
-		    sem_wait(&ackSemaphore);
+            for (i = 0; i < size - barSize; i++)
+                    sem_wait(&ackSemaphore);
             pthread_mutex_lock(&stateMut);
             sem_destroy(&ackSemaphore);
+}
+
+void waitAndChangeState( state_t state, int t)
+{
+	sleep(rand() % t);
+	if(state==InBar) pthread_mutex_lock(&stateMut);
```

Comment:
Według mnie lepiej przenieść pthread_mutex_lock do changeState, to będzie bardziej sensowne